config/travis/install.sh

```diff
@@ -32,36 +43,47 @@ then
 	docker exec ${CONTAINER_NAME} locale-gen en_US.UTF-8;
 
 	# Install packages.
-	DPKG_PACKAGES="git";
-
-	if test ${TARGET} = "pylint";
-	then
-		DPKG_PACKAGES="${DPKG_PACKAGES} python3-distutils pylint";
-	fi
-	DPKG_PACKAGES="${DPKG_PACKAGES} python3";
-
-	if test ${TARGET} = "pypi";
+	if test -n "${TOXENV}";
 	then
-		DPKG_PACKAGES="${DPKG_PACKAGES} python3-pip";
+		DPKG_PACKAGES="build-essential python${TRAVIS_PYTHON_VERSION} python${TRAVIS_PYTHON_VERSION}-dev tox";
 	else
-		DPKG_PACKAGES="${DPKG_PACKAGES} ${DPKG_PYTHON3_DEPENDENCIES} ${DPKG_PYTHON3_TEST_DEPENDENCIES}";
+		DPKG_PACKAGES="";
+
+		if test "${TARGET}" = "coverage";
+		then
+			DPKG_PACKAGES="${DPKG_PACKAGES} curl git";
+
+		elif test "${TARGET}" = "jenkins3";
+		then
+			DPKG_PACKAGES="${DPKG_PACKAGES} sudo";
+
+		elif test ${TARGET} = "pylint";
+		then
+			DPKG_PACKAGES="${DPKG_PACKAGES} python3-distutils pylint";
+		fi
+		if test ${TARGET} = "pypi";
+		then
+			DPKG_PACKAGES="${DPKG_PACKAGES} python3 python3-pip";
+
+		elif test "${TARGET}" != "jenkins3";
+		then
+			DPKG_PACKAGES="${DPKG_PACKAGES} python3 ${DPKG_PYTHON3_DEPENDENCIES} ${DPKG_PYTHON3_TEST_DEPENDENCIES}";
+
+			# Note that run_tests.py currently requires pylint.
+			DPKG_PACKAGES="${DPKG_PACKAGES} python3-distutils pylint";
+		fi
 	fi
 	docker exec -e "DEBIAN_FRONTEND=noninteractive" ${CONTAINER_NAME} sh -c "apt-get install -y ${DPKG_PACKAGES}";
 
 	docker cp ../timesketch ${CONTAINER_NAME}:/
 
-	if test ${TARGET} = "pypi";
+	if test "${TARGET}" = "pypi";
 	then
 		docker exec ${CONTAINER_NAME} sh -c "cd timesketch && pip3 install -r requirements.txt";
 		docker exec ${CONTAINER_NAME} sh -c "cd timesketch && pip3 install -r test_requirements.txt";
-		docker exec ${CONTAINER_NAME} sh -c "(cd timesketch/api_client/python && python3 setup.py build && python3 setup.py install)";
-		docker exec ${CONTAINER_NAME} sh -c "(cd timesketch/importer_client/python && python3 setup.py build && python3 setup.py install)";
+		# docker exec ${CONTAINER_NAME} sh -c "(cd timesketch/api_client/python && python3 setup.py build && python3 setup.py install)";
```

Comment:
Can you add a comment to explain the reason why these are disabled?

Reply:
They don't appear to be needed I can remove them as well